### PR TITLE
WT-9046 Don't call error handler when logging an error with __stdio_printf

### PR DESCRIPTION
**Pull request by Sean04**
Returning EIO from __stdio_printf if `vfprintf` fails (instead of `WT_ERR_MSG`), prevents the stack from overflowing with printf errors with printf is unable to print.

### CODE REVIEW

File: src/os_common/os_fstream_stdio.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ __stdio_printf(WT_SESSION_IMPL *session, WT_FSTREAM *fs, const char *fmt, va_lis
 {
```
Comment (luke-pearson) [inline src/os_common/os_fstream_stdio.c:50]: Please remove the session.

**Review by luke-pearson — CHANGES_REQUESTED** [review]

File: src/os_common/os_fstream_stdio.c (hunk 1 above)
Comment (Sean04) [reply src/os_common/os_fstream_stdio.c:50]: The session is used in other implementations of the `fstr_printf` function so I've just added a WT_UNUSED instead.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Review by ddanderson — APPROVED** [review]
LGTM

**Pull request merged**